test-build.sh

```diff
@@ -59,14 +62,13 @@ for version in "${versions[@]}"; do
   # See details in function.sh
   IFS=' ' read -ra variants <<< "$(get_variants "$(dirname "$version")" "${variant_arg[@]}")"
 
-  # Only build the default Dockerfile if "default" is in the variant list
-  if [[ "${variants[*]}" =~ "default" ]] || [[ "${variants[*]}" =~ "onbuild" ]] ; then
-    build "$version" "" "$tag"
-  fi
-
   for variant in "${variants[@]}"; do
     # Skip non-docker directories
-    [ -f "$version/$variant/Dockerfile" ] || continue
+    [ -f "$version/Dockerfile" ] || [ -a "$version/$default_variant/Dockerfile" ] || continue
+
+    if [[ "$variant" = "onbuild" ]]; then
```

Comment:
Maybe we don't need to use `[[ ]]` but just `[ ]` here?

Reply:
I can try